main.go

```diff
@@ -194,15 +194,40 @@ func updateMeasureLog() error {
 	log.Trace("main:updateMeasureLog() Entering")
 	defer log.Trace("main:updateMeasureLog() Leaving")
 
-	secLog.Infof("%s main:updateMeasureLog() Running %s using system administrative privileges", message.SU, constants.ModuleAnalysis)
-	cmd := exec.Command(constants.ModuleAnalysis)
-	cmd.Dir = constants.BinDir
-	results, err := cmd.Output()
+	secLog.Infof("%s main:updateMeasureLog() Running code to read EventLog", message.SU)
+	eventLogInfo := resource.GetEventLogInfo()
+
+	err := eventLogInfo.FetchUefiEventInfo()
+	if err != nil {
+		log.Errorf("main:updateMeasureLog() Error while getting UEFI Event Log Info: %s\n", err.Error())
+	} else {
+		err = eventLogInfo.UpdateUefiEventLog()
+		if err != nil {
+			log.Errorf("main:updateMeasureLog() Error while updating UEFI Event Log: %s\n", err.Error())
+		}
+	}
+
+	err = eventLogInfo.FetchTxtHeapInfo()
 	if err != nil {
-		return errors.Errorf("main:updateMeasureLog() module_analysis_sh error: %s", results)
+		log.Errorf("main:updateMeasureLog() Error while getting TXT Event Log Info: %s\n", err.Error())
+	} else {
+		eventLogInfo.TxtEnabled = true
+		err = eventLogInfo.UpdateTxtEventLog()
+		if err != nil {
+			log.Errorf("main:updateMeasureLog() Error while updating TXT Event Log: %s\n", err.Error())
+		}
+	}
+
+	err = eventLogInfo.UpdateAppEventLog()
+	if err != nil {
+		log.Errorf("main:updateMeasureLog() Error while updating Application Event Log: %s\n", err.Error())
+	}
+
+	if eventLogInfo.FinalPcrEventLog != nil {
+		eventLogInfo.WriteMeasureLogFile()
```

Comment:
I think I would have implemented this a little different but will defer to @ddhawal on what should be changed.  My recommendations are...

1. Create a new package/subfolder called eventlog ("resource" is intended for REST endpoints).
2. In that package, define a single "public" EventLogParser interface that has a one method "GetEventLogs() ( []PcrEventLog, error)".  Also expose data public data structures that have json decorations (ex. "PcrData").  
3. Add a free function (NewEventLogParser) that creates an instance of interface. 
4. Keep everything else internal to the package and encapsulate functions like "FetchUefiEventInfo()" and "FetchTxtHeapInfo()".  In other words, "main" would just news up the EventLogParser, calls "GetEventLogs" (returns the []PcrEventLog) and serializes that array to a contants.MeasureLogFilePath.

Reply:
I would go with recommendations from @kwtj43 , I had a hard time in understanding how does this come from resource

Reply:
@ddhawal 
 We have changed our structure according to Kent's comment.
 Now we have one package/folder called eventlog.
 In this we have two files
 1. event_log.go :
    This file contains only the interface function and public data structure
    i.e  GetEventLogs() and a free function (NewEventLogParser) that creates an instance of interface
 2. collect_event.go: This is a common file to collect all events(txt, uefi, application).       
 or 
 Do you want us to have separate files to collect each txt, uefi and application event ?
 Based on your comment we will create separate file like : collect_txt_event.go, collect_uefi_event.go, collect_application_event.go